src/interfaces/TypedData.ts

```diff
@@ -0,0 +1,21 @@
+export interface MessageTypeProperty {
+  name: string;
+  type: string;
+}
+
+export interface MessageTypes {
+  [additionalProperties: string]: MessageTypeProperty[];
+}
+
+export interface TypedMessage<T extends MessageTypes> {
+  types: T;
+  primaryType: keyof T;
+  domain: {
+    name?: string;
+    version?: string;
+    chainId?: number;
+    verifyingContract?: string;
+    salt?: ArrayBuffer;
```

Comment:
nit: why are all these fields optional? Wouldn't we either always set them or never set them for our implementation? Or am I missing something?

Reply:
Yea, in our implementation, we only set `name`, `version`, and `chainId`, but according to the spec, all these fields are optional. See https://eips.ethereum.org/EIPS/eip-712.

My initial goal was to make this specific type generic so that other typed data definitions could be implemented by it in the future.